document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresFunctionExpressionVisitor.java

```diff
@@ -0,0 +1,53 @@
+package org.hypertrace.core.documentstore.postgres.query.v1.vistors;
+
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.hypertrace.core.documentstore.expression.impl.FunctionExpression;
+import org.hypertrace.core.documentstore.expression.operators.FunctionOperator;
+
+public class PostgresFunctionExpressionVisitor extends PostgresSelectTypeExpressionVisitor {
+  @Override
+  public String visit(final FunctionExpression expression) {
+    int numArgs = expression.getOperands().size();
+    if (numArgs == 0) {
+      throw new IllegalArgumentException(
+          String.format("%s should have at least one operand", expression));
+    }
+
+    PostgresSelectTypeExpressionVisitor selectTypeExpressionVisitor =
+        new PostgresDataAccessorIdentifierExpressionVisitor(
+            new PostgresConstantExpressionVisitor());
+
+    if (numArgs == 1) {
+      String value = expression.getOperands().get(0).accept(selectTypeExpressionVisitor);
+      return String.format("%s( %s )", expression.getOperator().toString(), value);
+    }
+
+    Collector<String, ?, String> collector =
+        getCollectorForFunctionOperator(expression.getOperator());
+
+    String childList =
+        expression.getOperands().stream()
+            .map(exp -> exp.accept(selectTypeExpressionVisitor))
+            .filter(str -> !StringUtils.isEmpty((String) str))
+            .map(str -> "" + str + "")
+            .collect(collector);
```

Comment:
Nit: Can be simplified to
```suggestion
    String childList =
        expression.getOperands().stream()
            .map(exp -> exp.accept(selectTypeExpressionVisitor))
            .map(Object::toString)
            .filter(StringUtils::isNotEmpty)
            .collect(collector);
```

Reply:
If we also need to perform null checks, then
```suggestion
    String childList =
        expression.getOperands().stream()
            .map(exp -> exp.accept(selectTypeExpressionVisitor))
            .filter(Objects::nonNull)
            .map(Object::toString)
            .filter(StringUtils::isNotEmpty)
            .collect(collector);
```

Reply:
Done